### PR TITLE
Add High-score recording for Haiku2048

### PR DESCRIPTION
**Pull request by Oyami-Srk**
High-score record will be saved to Highscore.dat.

### CODE REVIEW

File: Game.cpp [hunk 1]
```diff
@@ -21,6 +23,16 @@ Game::Game(uint32 sizeX, uint32 sizeY)
 	fInGame(false),
 	fScore(0)
 {
+	std::ifstream highscore("Highscore.dat", std::ios::binary);
```
Comment (tuankiet65) [inline Game.cpp]: You should put Highscore.dat under B_USER_SETTINGS_DIRECTORY, if you do it like that then the file will end up at whatever current directory it is (even /system/apps, which is not good)

Example here: https://github.com/haikuports/haikuports/wiki/FindDirectory

**Comment by raefaldhia** [comment]
I think the name request event should be fired when the game ended (when you got the latest score).

**Comment by scottmc** [comment]
Note that other code in Haiku2048 just got merged and now conflicts with this.  Please check your code and see how much work would be involved with getting it to merge and still function as intended.

File: Game.h [hunk 2]
```diff
@@ -11,6 +11,9 @@
 #include <Looper.h>
 #include <SupportDefs.h>
 
+#define HIGHSCORE_FILENAME "Highscore.dat"
```
Comment (humdingerb) [inline Game.h]: We're not used to file suffixes in Haiku (at least for ususally not shared files). Just call it "Highscore".

File: Game.h [hunk 3]
```diff
@@ -11,6 +11,9 @@
...
+#define HIGHSCORE_DIRECTORY "Haiku2048"
```
Comment (humdingerb) [inline Game.h]: There'll probably be another (more?) setting file added in the future: window size/position, board state etc.
Just call it "Haiku2048_DIRECTORY".

File: TerminalBoard.cpp [hunk 4]
```diff
@@ -46,6 +46,12 @@ TerminalBoard::boardChanged(bool canUndo)
 	showBoard();
 }
 
+void
+TerminalBoard::nameRequest()
+{
+	std::cout << "Please fill the name and press enter key in window!" << std::endl;
```
Comment (humdingerb) [inline TerminalBoard.cpp]: Better: ""Please type in your name and press enter in the window."

File: WindowBoard.cpp [hunk 5]
```diff
@@ -176,6 +200,11 @@ GameWindow::showBoard(bool canUndo)
 	BString score;
 	score << "Score: " << fMaster->fTarget->Score();
 	fScore->SetText(score.String());
+	BString score_highest;
+	score_highest << "High Score: " << fMaster->fTarget->Score_Highest();
+	if(fMaster->fTarget->Username()[0])
+		score_highest << " By " << fMaster->fTarget->Username();
```
Comment (humdingerb) [inline WindowBoard.cpp]: Should be lower-case "by"

File: TerminalBoard.cpp [hunk 6]
```diff
@@ -118,5 +124,10 @@ TerminalBoard::showBoard()
 		}
 		std::cout << '\n';
 	}
-	std::cout << dashes << "\nScore: " << fTarget->Score() << std::endl;
+	std::cout << dashes << "\nHigh Score: " << fTarget->Score_Highest();
+	if(fTarget->Username()[0])
+		std::cout << " By " << fTarget->Username() << std::endl;
```
Comment (humdingerb) [inline TerminalBoard.cpp]: Should be lower-case "by".

**Comment by Oyami-Srk** [comment]
Thanks for the review. Those problems have been fixed.

**Pull request merged**